# Question 1: 

Given some sample data, write a program to answer the following: [click here to access the required data set](https://docs.google.com/spreadsheets/d/16i38oonuX1y1g7C_UAmiK9GkY7cS-64DfiDMNiR41LM/edit#gid=0)

On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis. 


## Exploring the Data

In [7]:
# import dataset
import pandas as pd
df = pd.read_csv("2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv")
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

In [13]:
# check the number of 30 day windows
df.sort_values(by=['created_at'])['created_at']

1862    2017-03-01 0:08:09
1741    2017-03-01 0:10:19
3228    2017-03-01 0:14:12
1267    2017-03-01 0:19:31
2689    2017-03-01 0:22:25
               ...        
2765    2017-03-30 9:22:41
4890    2017-03-30 9:27:00
244     2017-03-30 9:30:28
211     2017-03-30 9:40:40
145     2017-03-30 9:55:00
Name: created_at, Length: 5000, dtype: object

In [8]:
# summary of dataset
df.describe()

order_id      shop_id      user_id   order_amount  total_items
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720
std    1443.520003    29.006118    87.798982   41282.539349    116.32032
min       1.000000     1.000000   607.000000      90.000000      1.00000
25%    1250.750000    24.000000   775.000000     163.000000      1.00000
50%    2500.500000    50.000000   849.000000     284.000000      2.00000
75%    3750.250000    75.000000   925.000000     390.000000      3.00000
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000

In [11]:
# check for duplicated rows
df.duplicated().sum()

0

**Initial Analysis:** 
* The dataset pulled was over one 30-day window in March 2017
* There are exactly 5000 rows in the dataset
* There are exactly 100 distinct shop_id
* The mean for order_amount is 3145.128000
* There are no missing values
* There are no duplicated rows

**a.** Think about what could be going wrong with our calculation. Think about a better way to evaluate this data. 

The naive average order value (AOV) of 3145.13 was most likely the mean order_amount value which is sensitive to large outlier values in the dataset. In looking at the 50 percentile mark, the average looks to be around 284 which is much lower than the naive AOV. One possibility is that there may have been some errors in input from the shops or users in the order_amount which may be contributing to the large AOV. We can tell if this is the case by looking at the AOV per shop or user. 

**b**. What metric would you report for this dataset?

I would use other metrics such as median or mode as these metrics are less sensitive to outliers. In analyzing the data set, the values for median and mode were the same. However, as the data is from the 100 shops, the most frequently occuring sneaker price, mode, would provide a more accurate representation than AOV per shop. 

**c**. What is its value?

$153

# Question 2: 
For this question you’ll need to use SQL. [Follow this link](https://www.w3schools.com/SQL/TRYSQL.ASP?FILENAME=TRYSQL_SELECT_ALL) to access the data set required for the challenge. Please use queries to answer the following questions. Paste your queries along with your final numerical answers below.


**a.** How many orders were shipped by Speedy Express in total?

**Answer:** 54

**b.** What is the last name of the employee with the most orders?

**Answer:** Peacock

**c.** What product was ordered the most by customers in Germany?

**Answer:** Boston Crab Meat